kubernetes/structures_pod.go

```diff
@@ -107,6 +108,23 @@ func flattenPodSpec(in v1.PodSpec) ([]interface{}, error) {
 	}
 
 	if len(in.Volumes) > 0 {
+		// If the serviceAccountToken is being added to the list of volumes,
+		// we need to remove it prior to recording the state.
+		if in.AutomountServiceAccountToken != nil && *in.AutomountServiceAccountToken == true {
+			for i, volume := range in.Volumes {
+				nameMatchesDefaultToken, err := regexp.MatchString("default-token-([a-z0-9]{5})", volume.Name)
```

Comment:
What I would do is put this logic [here](https://github.com/hashicorp/terraform-provider-kubernetes/blob/master/kubernetes/structures_pod.go#L256) in the loop that expands the volumes and use a `continue` statement to skip it, rather than removing it here.